torch/csrc/jit/passes/peephole_list_idioms.cpp

```diff
@@ -21,17 +22,312 @@ c10::optional<size_t> normalizeIndex(int64_t index, size_t len) {
   }
 }
 
+// Refine from Value -> len of list
+// TODO: vector may be faster
+using ListRefinement = std::unordered_map<Value*, int64_t>;
+
+// The intersection of the refinements is the Value* which are in both
```

Comment:
Adding an example of when intersectRefinements are applied would help understanding.